src/List.mo

```diff
@@ -417,4 +418,9 @@ module {
   public func toVarArray<A>(xs : List<A>) : [var A] =
     Array.thaw<A>(toArray<A>(xs));
 
+  /// Create an iterator from a list.
+  public func toIter<A>(xs: List<A>) : Iter.Iter<A> {
+    toArray<A>(xs).vals();
```

Comment:
I'd prefer a direct iterator on the list itself than a detour via array.

Reply:
Thank you for your kind review, I have added an implementation of direct conversion from List to Iter.